Tickets/sitcom 784

boutigny opened the pull request:
I need this technote to be reviewed. Please note that there are 2 associated notebooks in the notebooks directory that need to be reviewed too.

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
 
-Add content here
-================
+   In the second part, we perform a detailed unvestigation of a peculiar event occuring on June 28th, 2023 around 1:08 am where a strong noise were reported during and after an elevation slew.
```
unvestigation --> investigation

boutigny replied:
done

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+
+For this study we have selected large amplitude azimuth et elevation slews occuring on December 21st, 2023.
+
+The following plot shows the acceleration along the x axis recorded by VMS accelerometer #3 over a 24 hour-period on 2023/12/21. The other accelerometers / axes show a similar behavior. 
```
Is this meant to show a typical behavior or an extraordinary case you found?

boutigny replied:
It is not an extraordinary case, I have seen few occurrences of such vibration events. In fact it would be interesting to make a global study but this is outside the scope of this technical note. 
The idea is more to say : "we see instances of shaking events and we have the tools ready to investigate them in case it occurs again with the glass mirror"

nsevilla replied:
OK! I think it would be good to have this clearly stated. So that the actual conclusion is that this exists, the tools are available, and a possible plan is to make a global study.

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+The start and the end of the slew are indicated by the vertical dashed lines. We can identify the vibrations generated during the acceleration and deceleration phases corresponding to an increase 
+of the TMA torque of +/- 500,000 N.m.
+During the TMA rotation at constant speed we see 4 vibration events lasting a few seconds. These vibrations are not associated to torque variations. We can also detect these events on the M1M3 
+angular acceleration around the z axis. The 2 other M1M3 accelerometers do not show significant variations.
```
Specify that the other two accelerometers mentioned here are the ones for the other axes.

boutigny replied:
I change it to: "The two other components of the angular acceleration, around x and y, do not show a significant signal associated to the vibration events."

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+
+   VMS acceleration along x for accelerometer #3 with TMA orientation superimposed. The start and the end of the slew are indicated by the vertical dashed lines.
+
+The following plot shows the acceleration along x, y and z axis recorded by accelerometer #2 (other accelerometers show the same behavior) and HP forces along the same axes.
```
accelerometer --> VMS accelerometer

boutigny replied:
done

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+Elevation slews
+---------------
+
+Finding data period with large amplitude elevation only slews is more difficult tahn for azimuth slew. We selected the following one: 
```
tahn --> than

nsevilla replied:
Is this a 'typical' example?

boutigny replied:
I wasn't able to find enough large elevation slews where the TMA is not moving in azimuth to make statistics, so it is a particular event.

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+   VMS acceleration along y for accelerometer #2, TMA Orientation, TMA torque, TMA speed, M1M3 angular acceleration around x, y and z. The start and the end of the slew are indicated
+   by the vertical dashed lines.
+
+During the slew and between the acceleration and decelartion phase, we see a few vibration occurences detected by the VMS
```
decelartion --> deceleration

boutigny replied:
done

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+
+An analysis of the VMS vibration similar to the one described for the azimuth slew shows that the most prominent glitches appear when the TMA elevation is around:
+[43.4 55.7 60.0 61.3 65.7] degrees. The same analysis on another elevation-only slew (2023-12023 0- seqNum: 205) shows glitches at: [52.7 53.9 55.3 58.1 60.1] degrees. Even if from these
+2 slews, it seems that some glitches appear around the same elvation values, it is impossibe to draw any definitive conclusion with such a low statistics.
```
elvation --> elevation; impossibe --> impossible

boutigny replied:
corrected

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+
+   Spectrograms of the signal recorded by the 3 VMS accelerometers along x, y and z.
+
+The glitches are seen as vertical bands covering the whole frequency range and are visible in mostof the spectrograms.
```
mostof --> most of

boutigny replied:
corrected

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+Analysis of VMS data when the TMA is tracking
+---------------------------------------------
+
+In order to complete this study, we have selected an event where the TMA is tracking after sleing to a target position.
```
sleing --> slewing

boutigny replied:
corrected

nsevilla commented on index.rst:
```diff
@@ -4,9 +4,326 @@ Monitor coupling in M1M3 VMS system and TMA drives together
+
+* dayObs: 20231221
+* seqNum: 535
+* duration: 42s
```
I always found curious that these tracking events have 42 seconds (instead of 30-ish). You can disregard this comment but just in case you had some insight.

boutigny replied:
My understanding is that you have to track during ~2s of settle time + 2x15s exposure + 2x2s reading +2x2s for the shutter to move (not sure about this one). This is ~40s, there is still 2s missing...

nsevilla reviewed (changes requested):
Added some questions and minor comments, also there are some in the notebooks through the NBviewer. One general thing I am missing is some kind of conclusion or next steps.

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #4.    ax.set(ylabel="accel (mg)", xlabel="Time", title=f"{vms_date} - Accel x_3")
</pre><p>Accel x_3 --&gt; VMS Accel x_3</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725433'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='17'/>

boutigny replied:
<p>Done</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<p>HP --&gt; hardpoint</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725436'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='19'/>

boutigny replied:
<p>Done</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #8.            label=f"Accel 2_{axis}",
</pre><p>Accel --&gt; VMS Accel</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725440'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

nsevilla replied:
<pre class="ql-syntax" spellcheck="false">Line #11.    ax[2].plot(df_azi.index, df_ele["actualTorque"], label="declination")
</pre><p>declination --&gt; Elevation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725453'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='43'/>

nsevilla replied:
<pre class="ql-syntax" spellcheck="false">Line #31.    fig.suptitle(f"{date} - seqNum: {seqNum}")
</pre><p>Maybe something more descriptive here, like M1M3 Hardpoint acceloremeter PSD or Hardpoint VMS PSD</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725457'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='44'/>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<p>Remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725443'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='26'/>

boutigny replied:
<p>Done</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<p>Describe this plot a little bit.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725445'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='30'/>

boutigny replied:
<p>It was just a plot to check that the low pass filter was working and should have been removed --&gt;  removed now</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<p>Describe this plot a little bit.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725446'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='32'/>

boutigny replied:
<p>Same thing for this one --&gt; removed</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #7.    print(len(df_azi), len(df_ele))
</pre><p>Remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725449'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='42'/>

boutigny replied:
<p>Done</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #7.    ax[1].plot(df_ele.index, df_ele["actualPosition"], label="Declination")
</pre><p>Declination --&gt; Elevation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725452'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='43'/>

boutigny replied:
<p>Argh ! I thought that I had corrected all of these. Thanks !</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation.ipynb:
```diff
@@ -0,0 +1,1197 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #15.    ax[3].plot(df_ele.index, df_ele["actualVelocity"], label="Declination")
</pre><p>Declination --&gt; Elevation</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724725455'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='43'/>

boutigny replied:
<p>Done</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation_strong_shaking-28-06-2023.ipynb:
```diff
@@ -0,0 +1,840 @@
+{
```
<p>notebbok --&gt; notebook</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724734134'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='0'/>

boutigny replied:
<p>Corrected</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation_strong_shaking-28-06-2023.ipynb:
```diff
@@ -0,0 +1,840 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #8.    vms_dir = os.path.join("/scratch/users/b/boutigny/vmsdata", year, month)
</pre><p>Substitute explicit path with vms_top_dir. Is there some generic place we can have this data that does not rely on your scratch space?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724734139'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='5'/>

boutigny replied:
<p>Unfortunately not. Petr was supposed to move these files to the LFA but he is way too busy to do it.</p>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation_strong_shaking-28-06-2023.ipynb:
```diff
@@ -0,0 +1,840 @@
+{
```
<p>HP --&gt; hardpoint</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724734141'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='21'/>

nsevilla commented on notebooks/SITCOMTN_131_M1M3_VMS_TMA_Coupling_investigation_strong_shaking-28-06-2023.ipynb:
```diff
@@ -0,0 +1,840 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #25.            print(t_peak)
</pre><p>Remove</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/lsst-sitcom/sitcomtn-131/pull/1/discussion/#comment-1724734143'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='RESOLVED' data-cellIndex='25'/>

boutigny replied:
<p>I think that it is interesting to have the peak positions printed out in order to find which ones are common to several sensors / axis. </p><p>I improved the format to make it nicer and clearer</p>

boutigny commented:
Thanks for the review, I think that I addressed all your comment. I also added a general conclusion in the TN saying the following:
_In this Technical Note, we have described a detailed investigation of two events where strong vibrations were detected by the VMS system in correlation with other sensors. We have developped the tools (notebooks) to identify and analyse these events and they are ready to be re-used during the next campaign of measurements with the glass mirrors. Unfortunately it was not possible to identify the exact source of these vibrations as the analysis was performed while the cell + surrogate mirror were dismounted. Using these notebooks during the future tests of the TMA will probably allow to identify potential problems and to react quickly._

nsevilla reviewed (approved):
Looks good, thanks
